tests/test_inner.py

```diff
@@ -779,9 +828,13 @@ def test_workflow_docker_build_error():
                                                   }}],
                                    plugin_files=[this_file])
 
-    assert workflow.build_docker_image().is_failed()
-
+    try:
+        workflow.build_docker_image()
+    except Exception as ex:
+        pass
```

Comment:
You want something like:
```
with pytest.raises(PluginFailedException):
    workflow.build_docker_image()
assert ...
```

i.e. just don't have the assert in the same block as the thing you expect to raise an exception (because it will escape that block).